### ISSUE TITLE
Story : Plots , Metrics for LLM support [2]

### ISSUE THREAD

**Issue by SoyGema** [iterative/dvc-render#138]
### Submission Type
- [ ] Discussion

### Context 
Offer ML support for another use cases beyond Translation, that could mostly imply text generation (Q/A , summarization, etc)
Separated from #137 in other to focus on solving plots and one use case and offer possible support for others.

### Impact 
- Offer support

### Issue creator Goal 
- Offer support

Leaving a placeholder here to possible questions. 
Will prob submit some q and a coming from discussion in the meantime
Thanks!
```[tasklist]
### Tasks
- [x] Define use case Q&A , metrics and plots for each one
- [x] Define use case Summarization , metrics and plots for each one
```

**Comment by SoyGema** [comment]
### Summarization 

#### Tasks variants

Summarization creates a shorter version of a document or an article that captures all the important information. Along with translation, it is another example of a task that can be formulated as a sequence-to-sequence task. Summarization can be:

- *Extractive*: extract the most relevant information from a document.
- *Abstractive*: generate new text that captures the most relevant information.

 I saw that @dberenbaum used this in his [example](https://github.com/dberenbaum/hf_seq2seq), so no much to add. Maybe he could use  some of this text above to complement his README.MD + add [html studio link](https://studio.iterative.ai/user/dberenbaum/projects/hf_seq2seq-eikvo7y9b7) to give it more reach in his repo ? If Dave read this at some point, please also consider changing the repo name to something more task oriented + frameworks, so it can be indexed as such when users search.  - _summarization_hf_dvc_ instead of _seq2seqhf_ - 
Thanks for making it anyway!


#### Code

Works with certain model architectures (BART, Pegasus, T5 family) . Uses Trainer class from HF Transformers library so DVCLive HF Callback can be easily implemented here . Last DVCLive HF taken from [#649](https://github.com/iterative/dvclive/pull/649)
Load dataset ,tokenizer and preprocessing function omited for simplicity but can be found [here](https://huggingface.co/docs/transformers/tasks/summarization)


```python

from dvclive.huggingface import DVCLiveCallback
from transformers import AutoModelForSeq2SeqLM, Seq2SeqTrainingArguments, Seq2SeqTrainer

model = AutoModelForSeq2SeqLM.from_pretrained("t5-small")
...
...
training_args = Seq2SeqTrainingArguments(
    output_dir="my_awesome_billsum_model",
    evaluation_strategy="epoch",
    learning_rate=2e-5,
    per_device_train_batch_size=16,
    per_device_eval_batch_size=16,
    weight_decay=0.01,
    save_total_limit=3,
    num_train_epochs=4,
    predict_with_generate=True,
    fp16=True,
    push_to_hub=True,
)

trainer = Seq2SeqTrainer(
    model=model,
    args=training_args,
    train_dataset=tokenized_billsum["train"],
    eval_dataset=tokenized_billsum["test"],
    tokenizer=tokenizer,
    data_collator=data_collator,
    compute_metrics=compute_metrics,
)

trainer.train()

```

#### Metrics
During training, the model computes the **loss**.
As evaluation metrics for the fine-tuned model , we use [ROUGE](https://huggingface.co/spaces/evaluate-metric/rouge) We define the metric, take the theoretical answer and the predicted answer, and compute the metric. This evaluation metric is computed **after fine-tuning**.

It returns a dictionary of the form that can be logged into dvc API 

```python
import evaluate
metric = evaluate.load("rouge")

```

As I understand, once the model is finetuned we should see these values in each table row (in DVC VSCode extension) per experiment. 

```python
with Live() as live:
    live.log_metric(("rouge", metric)

```
